app/service/rest_svc.py

```diff
@@ -401,27 +402,54 @@ async def _persist_ability(self, access, ab):
             - add parsers back in to current ability
             - save current ability to disk, then re-load ability from file
             - check/set executor timeouts on loaded abilities
+        :param access: Current access list
+        :type access: dict
+        :param ab: Ability to add or
+        :type ab: dict
+        :return: Created / updated ability
+        :rtype: List(Ability)
         """
-        if not ab.get('id') or not ab['id']:
+        # Set ability ID if undefined
+        if not ab.get('id'):
             ab['id'] = str(uuid.uuid4())
+
+        if not self.is_uuid4(ab.get('id')):
+            self.log.debug('Invalid ability ID "{}". ID must be UUID4.'.format(ab.get('id')))
+            return []
```

Comment:
update id check to min required

Reply:
Updated to alphanumeric, hyphens, underscores. All current abilities use lowercase alpha, numeric, and hyphens exclusively. Had trouble including periods since the id is used for jQuery selectors in a lot of places.